include/boost/stacktrace/detail/mgwhelp_impls.hpp

```diff
@@ -0,0 +1,136 @@
+#ifndef BOOST_STACKTRACE_DETAIL_MGWHELP_IMPLS_HPP
+#define BOOST_STACKTRACE_DETAIL_MGWHELP_IMPLS_HPP
+
+#include <boost/config.hpp>
+#ifdef BOOST_HAS_PRAGMA_ONCE
+#   pragma once
+#endif
+
+#include <boost/stacktrace/frame.hpp>
+#include <boost/detail/winapi/get_current_process.hpp>
+#include <windows.h>
+#include <dbghelp.h>
+#include <Shlobj.h>
+
+namespace boost { namespace stacktrace { namespace detail {
+
+bool initialize_sym(HANDLE process_handle) {
+    std::string sym_search_path;
+
+    if( (getenv("_NT_SYMBOL_PATH") == NULL) && (getenv("_NT_ALTERNATE_SYMBOL_PATH") == NULL)) {
+        char local_app_data[MAX_PATH];
+        HRESULT hr = SHGetFolderPathA(NULL, CSIDL_LOCAL_APPDATA, NULL, 0, local_app_data);
+        if (SUCCEEDED(hr))
+            sym_search_path = std::string("srv*") + std::string(local_app_data) + std::string("\\drmingw*http://msdl.microsoft.com/download/symbols");
+        else
+            sym_search_path = "srv*http://msdl.microsoft.com/download/symbols";
+    }
+
+    return SymInitialize(process_handle, sym_search_path.c_str(), true);
```

Comment:
Please prepend all the system calls and structures with `::` to disable ADL and make sure that the right function/class is used.